book/chapters/chapter10/advanced_technical_aspects_of_mlr3.qmd

```diff
@@ -78,12 +75,17 @@ system.time({parSapply(cl, x, f, chunk.size = 1)})
 system.time({parSapply(cl, x, f, chunk.size = 2500)})
 ```
 
-Whenever you have the option to control the granularity by setting the chunk size, you should aim for at least as many jobs as workers and also the runtime of each worker should be at least several seconds.
-This ensures that you can fully utilize the system and that the parallelization overhead stays reasonable.
+Whenever you have the option to control the granularity by setting the chunk size, you should aim for at least as many jobs as workers.
+However, if there are too few job chunks with strongly dissimilar runtimes, the system may end up waiting for the last chunk to finish, while other ressources are idle.
+This is referred to as `r index("synchronization overhead", aside = TRUE)`.
+You should therefore aim for chunks with a runtime of at least several seconds, so that the parallelization overhead remains reasonable, while still having enough chunks to ensures that you can fully utilize the system.
```

Comment:
```suggestion
You should therefore aim for chunks with a runtime of at least several seconds, so that the parallelization overhead remains reasonable, while still having enough chunks to ensure that you can fully utilize the system.
```